core/src/main/scala/org/apache/spark/deploy/history/ApplicationCache.scala

```diff
@@ -279,7 +279,8 @@ private[history] class ApplicationCache(
     time(metrics.loadTimer) {
       operations.getAppUI(appId, attemptId) match {
         case Some(LoadedAppUI(ui, updateState)) =>
-          val completed = ui.getApplicationInfoList.exists(_.attempts.last.completed)
+          val completed = ui.getApplicationInfo(appId)
+            .map(_.attempts.last.completed).getOrElse(false)
```

Comment:
Though I get what you're trying to do here, I'm not sure this is actually faster in this case since the list is only length one. It may be cleaner to leave this as is, what are other reviewers opinions?